docs/source/notebooks/wps_generate_climos.ipynb

```diff
@@ -0,0 +1,388 @@
+      "    </metaurl>\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Process dry run output\n",
+    "req = requests.get(dry_output.get()[0])\n",
+    "soup = BeautifulSoup(BeautifulSoup(req.content.decode('utf-8')).prettify()).metaurl\n",
+    "print(soup)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Test dry run output is as expected\n",
+    "url = soup.get_text()\n",
+    "with urlopen(url) as f:\n",
+    "    output_data = f.read().decode('utf-8')\n",
+    "\n",
+    "expected_output = '''Dry Run\n",
+    "generate_climos:\n",
+    "INFO:dp.generate_climos:Processing: https://docker-dev03.pcic.uvic.ca/twitcher/ows/proxy/thredds/dodsC/datasets/TestData/fdd_seasonal_CanESM2_rcp85_r1i1p1_1951-2100.nc\n",
+    "INFO:dp.generate_climos:climo_periods: {'6190'}\n",
+    "INFO:dp.generate_climos:project: CMIP5\n",
+    "INFO:dp.generate_climos:institution: PCIC\n",
+    "INFO:dp.generate_climos:model: CanESM2\n",
+    "INFO:dp.generate_climos:emissions: historical, rcp85\n",
+    "INFO:dp.generate_climos:run: r1i1p1\n",
+    "INFO:dp.generate_climos:dependent_varnames: ['fdd']\n",
+    "INFO:dp.generate_climos:time_resolution: seasonal\n",
+    "INFO:dp.generate_climos:is_multi_year_mean: False\\n'''\n",
+    "\n",
+    "assert output_data == expected_output"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# generate climos\n",
+    "output = thunderbird.generate_climos(\n",
+    "    netcdf=seasonal_opendap, \n",
+    "    operation=operation, \n",
+    "    climo=climo, \n",
+    "    resolutions=resolutions, \n",
+    "    dry_run=False\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "<metaurl mediatype=\"application/x-netcdf\">\n",
+      "     http://localhost:5001/outputs/5657bfa8-0368-11eb-b7e1-c9fa1b0c3983/fdd_aClimMean_BCCAQ_CanESM2_historical+rcp85_r1i1p1_19610101-19901231_Canada.nc\n",
+      "    </metaurl>\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Process normal output\n",
+    "req = requests.get(output.get()[0])\n",
+    "soup = BeautifulSoup(BeautifulSoup(req.content.decode('utf-8')).prettify()).metaurl\n",
+    "print(soup)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "<ipython-input-9-e3c8cb77b746>:9: DeprecationWarning: tostring() is deprecated. Use tobytes() instead.\n",
+      "  for key in input_data.__dict__:\n",
+      "<ipython-input-9-e3c8cb77b746>:12: DeprecationWarning: tostring() is deprecated. Use tobytes() instead.\n",
+      "  assert input_data.__dict__[key] == output_data.__dict__[key]\n"
```

Comment:
The DeprecationWarning we see here causes a fail in `pytest`. It is probably coming from one of the packages used in `generate_climos`.

Reply:
Ah okay. I will try and fix that. I think it is coming from the dictionary I'm using, I wasn't sure if it was a big deal.

Reply:
If the warning is coming from `ce-dataprep` you will need to manually run `generate_climos` and see which package in `requirements.txt` is outdated. I think a version release is required to resolve this.

Reply:
I removed this section of code and replaced it, and all the tests are passing now, without warnings